### PR TITLE
Clamp TabController animation value when calc _DragAnimation.value

### PR DESCRIPTION
**Pull request by smadey**
## Description

see [demo](https://dartpad.dev/cf3e344cfdb5aa08a17c0d36aeaa0681)

## Tests

I added the following tests:

*TabController changes offset attribute*

## Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

No

### CODE REVIEW

**Review by Piinks — COMMENTED** [review]
I think this is meant to resolve https://github.com/flutter/flutter/issues/61156
@smadey is this the bug your demo is illustrating? I wasn't sure what this is supposed to be fixing.

**Comment by smadey** [comment]
Not the same as #61156
See [gist](https://gist.github.com/smadey/cf3e344cfdb5aa08a17c0d36aeaa0681) or [repo](https://github.com/smadey/nest_tabs_demo)
Swipe left and right on the last outer tab and last inner tab to see the problem

Simulator Screen Shot - iPhone X - 2020-07-20 at 20 13 35

**Comment by Piinks** [comment]
> Not the same as #61156
> See [gist](https://gist.github.com/smadey/cf3e344cfdb5aa08a17c0d36aeaa0681) or [repo](https://github.com/smadey/nest_tabs_demo)
> Swipe left and right on the last outer tab and last inner tab to see the problem
> 

Ah! Ok thanks for checking!
Can you add a short description to your PR explaining the bug? It's a bit hard without guidance to tell from the picture what the bug is that you are fixing or how to reproduce it from your sample code. Thanks! :)

**Comment by smadey** [comment]
Swipe left and right on the last outer tab and last inner tab to see the problem
RPReplay_Final1596186704

File: packages/flutter/test/material/tabs_test.dart [hunk 1]
```diff
@@ -2670,6 +2670,57 @@ void main() {
     final PageView pageView = tester.widget<PageView>(find.byType(PageView));
     expect(pageView.physics.toString().contains('ClampingScrollPhysics'), isFalse);
   });
+
```
Comment (Piinks) [inline packages/flutter/test/material/tabs_test.dart:2673]: The analyzer is failing here due to a trailing whitespace.

**Review by Piinks — COMMENTED** [review]
It looks like there is a merge conflict, can you update your branch?

**Review by Piinks — APPROVED** [review]
This LGTM! Thank you for the contribution! 
It looks like there is a trailing whitespace that is causing the analyzer to fail. Once that's fixed we can merge this! 🎉

**Comment by Piinks** [comment]
Hey @smadey would you be willing to update your branch once more? I think that will resolve the failing check here.

**Comment by smadey** [comment]
close by #64715

**Pull request closed without merging**